app/models/concerns/pxe_loader_support.rb

```diff
@@ -0,0 +1,55 @@
+module PxeLoaderSupport
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def all_loaders_map(precision = 'x64')
+      {
+        "None" => "NONE",
+        "PXELinux BIOS" => "pxelinux.0",
+        "PXELinux UEFI" => "pxelinux.efi",
+        "Grub UEFI" => "grub/boot#{precision}.efi",
+        "Grub UEFI SecureBoot" => "grub/shim.efi",
+        "Grub2 UEFI" => "grub2/grub#{precision}.efi",
+        "Grub2 UEFI SecureBoot" => "grub2/shim.efi"
+      }.freeze
+    end
+
+    def all_loaders
+      all_loaders_map.keys.freeze
+    end
+  end
+
+  def available_loaders
+    self.class.all_loaders
```

Comment:
Not sure what's the purpose of this method, why not defining

```ruby
# Implemented only in the OSs subclasses where it makes sense, defaults to none
def available_loaders
  []
end
```

and then you just define it on the classes where it makes sense